lib/cli/src/commands/create_exe.rs

```diff
@@ -446,3 +534,195 @@ impl LinkCode {
+mod http_fetch {
+    use anyhow::{anyhow, Context, Result};
+    use http_req::{
+        request::Request,
+        response::{Response, StatusCode},
+        uri::Uri,
+    };
+    use std::convert::TryFrom;
+
+    pub fn get_latest_release() -> Result<serde_json::Value> {
+        let mut writer = Vec::new();
+        let uri = Uri::try_from("https://api.github.com/repos/wasmerio/wasmer/releases").unwrap();
+
+        let response = Request::new(&uri)
+            .header("User-Agent", "wasmer.io self update")
+            .header("Accept", "application/vnd.github.v3+json")
+            .timeout(Some(std::time::Duration::new(30, 0)))
+            .send(&mut writer)
+            .map_err(anyhow::Error::new)
+            .context("Could not lookup wasmer repository on Github.")?;
+
+        assert_eq!(response.status_code(), StatusCode::new(200));
+
+        let v: std::result::Result<serde_json::Value, _> = serde_json::from_reader(&*writer);
+        let mut response = v.map_err(anyhow::Error::new)?;
+        eprintln!("response: {:?}", &response);
+        if let Some(releases) = response.as_array_mut() {
+            releases.retain(|r| {
+                r["tag_name"].is_string() && !r["tag_name"].as_str().unwrap().is_empty()
+            });
+            releases.sort_by_cached_key(|r| r["tag_name"].as_str().unwrap_or_default().to_string());
+            if let Some(latest) = releases.pop() {
+                return Ok(latest);
+            }
+        }
+
+        Err(anyhow!(
+            "Could not get Github API response, falling back to downloading latest version."
+        ))
+    }
+
+    pub fn download_release(
+        mut latest: serde_json::Value,
+        target_triple: Option<wasmer::Triple>,
+    ) -> Result<String> {
+        eprintln!("latest: {:?}", &latest);
+        let target_triple = target_triple.unwrap_or(wasmer::Triple::host());
+        if let Some(assets) = latest["assets"].as_array_mut() {
+            assets.retain(|a| {
+                if let Some(name) = a["name"].as_str() {
+                    match target_triple.architecture {
+                        wasmer_types::Architecture::X86_64 => {
+                            name.contains("x86_64") || name.contains("amd64")
+                        }
+                        wasmer_types::Architecture::Aarch64(
+                            wasmer_types::Aarch64Architecture::Aarch64,
+                        ) => name.contains("arm64") || name.contains("aarch64"),
+                        _ => false,
+                    }
+                } else {
+                    false
+                }
+            });
+            assets.retain(|a| {
+                if let Some(name) = a["name"].as_str() {
+                    match target_triple.vendor {
+                        wasmer_types::Vendor::Apple => !name.contains("windows"),
+                        wasmer_types::Vendor::Pc => !name.contains("apple"),
+                        _ => true,
```

Comment:
Ummm... this seems somewhat incorrect?

What are we computing here? It's a bit confusing for me

Reply:
It is confusing yeah. Apple vendor does not necessarily mean that the given triple includes the vendor name, because `zig` has its own triple format and it doesn't include the vendor (for example `x86_64-macos-none` is a valid zig triple but in llvm/clang it's `x86_64-apple-darwin`). So the checks here are reversed because they can only single out artifacts, not find them all.

Reply:
What I mean is... what is the true/false value signaling for? I still have no idea. It's true that is Linux?

Reply:
If the vendor is Apple, it should definitely not include substring "windows". If the vendor is Pc, it should definitely not include "apple". In any other case just retain it anyways.